jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -208,8 +208,7 @@ public void init() throws ServletException
         if (contentFactory == null)
         {
             MimeTypes mimeTypes = _contextHandler.getMimeTypes();
-            ResourceFactory resourceFactory = baseResource != null ? ResourceFactory.of(baseResource) : this::getResource;
-            contentFactory = new ResourceHttpContentFactory(resourceFactory, mimeTypes);
+            contentFactory = new ResourceHttpContentFactory(Objects.requireNonNullElse(baseResource, _contextHandler.getBaseResource()), mimeTypes);
```

Comment:
I don't think `baseResource` can be null here.

Reply:
mwokay... but the previous code thought it could be.   I've removed and we will see what breaks.